bundles/org.openhab.ui/web/src/pages/settings/things/add/choose-thing-type.vue

```diff
@@ -224,29 +224,99 @@ export default {
         Promise.reject('Failed to load inbox: ' + e)
       })
     },
-    approve (entry) {
+
+    openEntryActions (entry) {
       console.log(`Add ${entry.thingUID} as thing`)
-      this.$f7.dialog.prompt(`This will create a new Thing of type ${entry.thingTypeUID} with the following name:`,
-        'Add as Thing',
-        (name) => {
-          this.$oh.api.postPlain(`/rest/inbox/${entry.thingUID}/approve`, name).then((res) => {
-            this.$f7.toast.create({
-              text: 'Entry approved',
-              destroyOnClose: true,
-              closeTimeout: 2000
-            }).open()
-            setTimeout(() => { this.$f7router.navigate('/settings/things/', { reloadCurrent: true }) }, 300)
-          }).catch((err) => {
-            this.$f7.toast.create({
-              text: 'Error during thing creation: ' + err,
-              destroyOnClose: true,
-              closeTimeout: 2000
-            }).open()
-          })
-        },
-        null,
-        entry.label)
+      let actions = this.$f7.actions.create({
+        convertToPopover: true,
+        closeOnEscape: true,
+        buttons: [
+          [
+            {
+              text: entry.label,
+              label: true
+            }
+          ],
+          [
+            {
+              text: 'Add as Thing',
+              color: 'green',
+              bold: true,
+              onClick: () => {
+                this.$f7.dialog.prompt(`This will create a new Thing of type ${entry.thingTypeUID} with the following name:`,
+                  'Add as Thing',
+                  (name) => {
+                    this.approveEntry(entry, name)
+                  },
+                  null,
+                  entry.label)
+              }
+            },
+            {
+              text: 'Add as Thing (with custom ID)',
+              color: 'blue',
+              bold: true,
+              onClick: () => {
+                this.$f7.dialog.prompt(`This will create a new Thing of type ${entry.thingTypeUID}. You can change the suggested thing ID below:`,
+                  'Add as Thing',
+                  (newThingId) => {
+                    this.$f7.dialog.prompt('Enter the desired name of the new Thing:',
+                      'Add as Thing',
+                      (name) => {
+                        this.approveEntry(entry, name, newThingId)
+                      },
+                      null,
+                      entry.label)
+                  },
+                  null,
+                  entry.thingUID.substring(entry.thingUID.lastIndexOf(':') + 1))
+              }
+            },
+            {
+              text: 'Show equivalent Thing file configuration (basic properties)',
```

Comment:
Can you please remove this from this PR? See https://github.com/openhab/openhab-webui/pull/2920#pullrequestreview-2519664477.

I would also suggest to refactor the `openEntryActions` and `approveEntry` actions to a mixin, which then both `inbox-list.vue` and `choose-thing-type.vue` can import. This reduces code duplication and avoids further issues where those two provide different options.